lib/panda/http_response.rb

```diff
@@ -21,9 +21,10 @@ def parse(body)
 
       @parsed_body = JSON.parse(body)
       @request_id = parsed_body['request_id']
-      return if parsed_body['code'] == SUCCESS_CODE
+      code = parsed_body['code']
+      return if code == SUCCESS_CODE
 
-      raise Panda::APIError.new(status, body)
+      raise Panda::ErrorHandler.new(code).error_class.new(status, body)
```

Comment:
This chain call doesn't look clean, I suggest to add class method to `Panda::ErrorHandler` (`handle` for ex.) that takes `code`, `status` and `body` and initialise and return appropriate error class.

Reply:
@xalmar example from close gem https://github.com/taylorbrooks/closeio/blob/master/lib/closeio/error.rb